back/api/views.py

```diff
@@ -34,13 +34,24 @@ def getRoutes(request):
     ]
     return Response(routes)
 
-class RegisterView(APIView):
+class UserRegisterView(APIView):
     def post(self, request):
         serializer = UserSerializer(data=request.data)
         serializer.is_valid(raise_exception=True)
         serializer.save()
         return Response(serializer.data)
 
+class TeacherRegisterView(APIView):
+    def post(self, request):
+        user = UserSerializer(data=request.data)
+        user.is_valid(raise_exception=True)
+        user.save()
+        user_id = User.objects.get(username=user.data['username']).id
+        serializer = TeacherSerializer(data={'user':user_id})
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data)
```

Comment:
Is this request create a Teacher instance or simply a User ..?

Reply:
To be explicit I renamed it "teacher_serializer"